org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -4896,6 +4897,10 @@ protected void validateState(IEditorInput input) {
 
 		} catch (CoreException x) {
 			IStatus status= x.getStatus();
+			if (status != null && status.getSeverity() == IStatus.CANCEL) {
```

Comment:
Maybe an IStatus.checkCancel() throws OperationCanceledException would be useful instead of coding checks explicitly.

Reply:
i see a lot getSeverity() == IStatus.CANCEL in the codebase but with very various implementations.

Reply:
One more reason for me to unify this.

Reply:
feel free to make another ticket for that. but i do not see a way how todo that.